doc/PLUGINS.md

```diff
@@ -514,14 +514,49 @@ It is currently extremely restricted:
    commands, as these may become intermingled and break rule #1.
 3. the hook will be called before your plugin is initialized!
 
+This hook, unlike all the other hooks, is also strongly synchronous:
+`lightningd` will stop almost all the other processing until this
+hook responds.
+
 ```json
 {
+  "data_version": 42,
   "writes": [
     "PRAGMA foreign_keys = ON"
   ]
 }
 ```
 
+This hook is intended for creating continuous backups.
+The intent is that your backup plugin maintains two files:
+(1) an (obsolete) copy of the database, and (2) a file of
+database queries that will bring that database up-to-date.
+
+`data_version` is an unsigned 32-bit number that will always
+increment by 1 each time `db_write` is called.
```

Comment:
This is technically not true, we will wrap around at 2**32-1, but for most people this is sufficient I guess.

Reply:
Added line about wraparound for unsigned 32-bit ints.